### PR TITLE
[ADD] account_customer_wallet: Search on customer_wallet_balance

### PR DESCRIPTION
**Pull request by carmenbianca**
## Odoo task (if applicable)

https://gestion.coopiteasy.be/web#id=8557&view_type=form&model=project.task

## Checklist before approval

- [x] Tests are present (or not needed).
- [x] Credits/copyright have been changed correctly.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: account_customer_wallet/models/res_partner.py [hunk 1]
```diff
@@ -71,3 +73,23 @@ def _compute_customer_wallet_balance(self):
...
+                operator = "=="
+            filter_string = "partner.customer_wallet_balance {operator} {value}".format(
+                operator=operator, value=value
+            )
+            filtered = self.search([]).filtered(
+                lambda partner: safe_eval(
+                    filter_string,
+                    {"partner": partner},
+                )
+            )
```
Comment (victor-champonnois) [inline account_customer_wallet/models/res_partner.py:90]: Wouldn't it be faster (although more risky?) to it with an SQL query ?

**Review by victor-champonnois — COMMENTED** [review]
Just a question, otherwise, LGTM. Also I learned about the `safe_eval` method :smile: ! FYI I have a similar PR here https://github.com/OCA/partner-contact/pull/1287. If you have the time I would appreciate your point of view on it.

**Review by victor-champonnois — APPROVED** [review]

File: account_customer_wallet/models/res_partner.py (hunk 1 above)
Comment (carmenbianca) [reply account_customer_wallet/models/res_partner.py:90]: That would require me to be better at writing SQL than I am ;)

But yes, you're right.

**Pull request merged**